hawc/apps/epiv2/mixins.py

```diff
@@ -0,0 +1,75 @@
+from rest_framework import serializers
+
+from django.db import models
+
+from .models import Design
+
+
+class SameDesignSerializerMixin:
+    """
+    ex: when updating an ExposureLevel, there's a related Chemical and Exposure sometimes passed into the payload as
+    "chemical_id" and "exposure_level_id". Basic DRF functionality verifies that the supplied values are valid
+    items in the database, but it's a common requirement that they also belong to the same parent object; in this
+    case, the same design. This mixin performs that type of validation logic.
+
+    Could potentially generalize this even further (so that it's not just useful for "Design"); this type of
+    relationship is not uncommon.
+    """
+
+    _fields_to_check: list[tuple[str, type[models.Model]]]
```

Comment:
I think this is a public attribute since you need to specify it on the class that uses it; remove underscore?

Reply:
reworked. re: formatting, I had been using flake8, didn't realize you'd switched to ruff. When I add that to my setup it catches everything (I think!)